packages/aws-appsync/src/helpers/offline.ts

```diff
@@ -252,18 +257,28 @@ const isObject = (object) => object != null && (typeof object === 'object')
  * 
  * @returns Mutation options to be used by the ApolloClient
  */
-const buildMutation = (
+const buildMutation = <T = OperationVariables>(
     client: ApolloClient<any>,
     mutation: DocumentNode,
-    variables: object = {},
+    variablesInfo: VariablesInfo<T> | T,
     cacheUpdateQuery: CacheUpdatesOptions,
     typename: string,
     idField: string = 'id',
     operationType?: CacheOperationTypes
 ): MutationOptions => {
+    const isVariablesInfo = typeof (variablesInfo as VariablesInfo).variables === 'object';
+    const variables = isVariablesInfo ? (variablesInfo as VariablesInfo).variables : variablesInfo as T;
+
+    const hasInputType = Object.keys(variables).length === 1 && typeof variables.input === 'object';
```

Comment:
Need to document that the key must be named `input` for customers that might name GraphQL input types something else.